src/snowflake/snowpark/modin/pandas/base.py

```diff
@@ -442,19 +442,29 @@ def _binary_op(
         # pandas itself will ignore the axis argument when using Series.<op>.
         # Per default, it is set to axis=0. However, for the case of a Series interacting with
         # a DataFrame the behavior is axis=1. Manually check here for this case and adjust the axis.
+        lhs_series_rhs_dataframe = False
         if isinstance(self, pd.Series) and isinstance(other, pd.DataFrame):
-            axis = 1
+            lhs_series_rhs_dataframe = True
 
         new_query_compiler = self._query_compiler.binary_op(
             op=op,
             other=other,
-            axis=axis,
+            axis=1 if lhs_series_rhs_dataframe else axis,
             level=level,
             fill_value=fill_value,
             squeeze_self=squeeze_self,
             **kwargs,
         )
-        return self._create_or_update_from_compiler(new_query_compiler)
+
+        from snowflake.snowpark.modin.pandas.dataframe import DataFrame
+
+        # Modin Bug: https://github.com/modin-project/modin/issues/7236
```

Comment:
will this Modin Bug be resolved in a future release? would this comment then be obsolete?

Reply:
This function varies quite significantly from open-source Modin, so let's keep the issue in for now as I'm unsure when Modin will resolve this.

Reply:
ah ok, thanks for clarifying. Should we open a ticket to keep track of reducing conflicts to Modin for this? I.e., add a TODO with SNOW ticket to ensure we do not forget about the deviation here?

Reply:
We have this comment at the top of the function `# TODO: SNOW-1119855: Modin upgrade - modin.pandas.base.BasePandasDataset` which is being used as a tag to see what needs to be upstreamed/migrated so this should be covered.